### PR TITLE
feat(sumo): Send complete VehicleType with ScenarioVehicleRegistration

### PR DESCRIPTION
**Pull request by kschrab**
## Description

* With this PR, the `ScenarioVehicleRegistration` not provides a complete `VehicleType`. This allows easier handling of vehicle types in mapping and application simulator, especially if no prototype is configured in the mapping_config.json for vehicles created by the simulator (e.g. via SUMOs rou-file).
* Accordingly, SumoAmbassador now creates a complete VehicleType for each vehicle defined in the rou-file based on vehicle parameters read via TraCI or Libsumo. 
* #289 could profit from this change.

## Issue(s) related to this PR

* Resolves internal issue 556
  
## Affected parts of the online documentation

Changes in the documentation required?

No

## Definition of Done

**Prerequisites**

- [x] You have read CONTRIBUTING.md carefully.
- [x] You have signed the [Contributor License Agreement](http://www.eclipse.org/legal/CLA.php).
- [x] Your GitHub user id is linked with your Eclipse Account.

**Required**

- [x] The title of this merge request follows the scheme `type(scope): description`  (in the style of [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary))
- [x] You have assigned a suitable label to this pull request (e.g., `enhancement`, or `bugfix`)
- [x] `origin/main` has been merged into your Fork.
- [x] Coding guidelines have been followed (see CONTRIBUTING.md).
- [x] All checks on GitHub pass.
- [x] All tests on [Jenkins](https://ci.eclipse.org/mosaic/job/mosaic/) pass.

**Requested** (can be enforced by maintainers)

- [x] New functionality is covered by unit tests or integration tests. Code coverage must not decrease.
- [ ] If a bug has been fixed, a new unit test has been written (beforehand) to prove misbehavior
- [x] There are no new SpotBugs warnings.

### CODE REVIEW

File: fed/mosaic-mapping/src/main/java/org/eclipse/mosaic/fed/mapping/ambassador/MappingAmbassador.java [hunk 1]
```diff
@@ -124,39 +121,37 @@ private void handleInteraction(ScenarioTrafficLightRegistration interaction) {
...
                         "This scenario vehicle \"{}\" of prototype \"{}\" will not be equipped due to a weight condition of {}.",
-                        scenarioVehicle.getVehicleTypeId(),
+                        scenarioVehicle.getVehicleType().getName(),
                         scenarioVehicle.getId(),
                         prototype.weight
                 );
-                applications = new ArrayList<>();
-                group = null;
+                return;
             }
```
Comment (kschrab) [inline fed/mosaic-mapping/src/main/java/org/eclipse/mosaic/fed/mapping/ambassador/MappingAmbassador.java:142]: @schwepmo I reverted your change and we now again send `VehicleRegistration` for true matches only. `ApplicationAmbassador` is now able to process `ScenarioVehicleRegistration` where it extracts the `VehicleType` and passes it to the perception index.

**Review by schwepmo — COMMENTED** [review]
Just one question and I removed some unnecessary code. Apart from that all looks good!

File: fed/mosaic-sumo/src/main/java/org/eclipse/mosaic/fed/sumo/bridge/facades/VehicleFacade.java [hunk 2]
```diff
@@ -152,6 +192,43 @@ public String getVehicleTypeId(String vehicleId) throws InternalFederateExceptio
...
+                        SumoVehicleClassMapping.fromSumo(getVehicleTypeVClass.execute(bridge, vehicleTypeId)),
+                        getVehicleTypeAccel.execute(bridge, vehicleTypeId),
+                        getVehicleTypeDecel.execute(bridge, vehicleTypeId),
+                        null,
+                        getVehicleTypeSigma.execute(bridge, vehicleTypeId),
+                        getVehicleTypeTau.execute(bridge, vehicleTypeId),
+                        getVehicleTypeSpeedFactor.execute(bridge, vehicleTypeId),
+                        null,
+                        null,
+                        null
```
[the remark this answers is not in the document]
Comment (kschrab) [reply fed/mosaic-sumo/src/main/java/org/eclipse/mosaic/fed/sumo/bridge/facades/VehicleFacade.java]: I added some clarification to the fields why we cannot read them

**Comment by kschrab** [comment]
@schwepmo Libsumo currently does not work on Windows with SUMO 1.16.0. libsumojni.dll cannot be loaded :( I raised an issue here: https://github.com/eclipse/sumo/issues/12605

However, I tried out Barnim with SUMO 1.14.0 and 1.15.0 using the 1.16.0 jarfile and it succeeded, therefore I added 1.14.0 and 1.15.0 to the compatible versions in `LibSumoAmbassador.VALID_LIBSUMO_VERSIONS`

File: fed/mosaic-mapping/src/main/java/org/eclipse/mosaic/fed/mapping/ambassador/MappingAmbassador.java (hunk 1 above)
Comment (kschrab) [reply fed/mosaic-mapping/src/main/java/org/eclipse/mosaic/fed/mapping/ambassador/MappingAmbassador.java]: @schwepmo are you fine with this specfic change?

Comment (schwepmo) [reply fed/mosaic-mapping/src/main/java/org/eclipse/mosaic/fed/mapping/ambassador/MappingAmbassador.java]: Yes, for me this works fine. I'm not sure if we need to adjust the `runtime.json` in MOSAIC extended as well?

Comment (kschrab) [reply fed/mosaic-mapping/src/main/java/org/eclipse/mosaic/fed/mapping/ambassador/MappingAmbassador.java]: That's part of another merge request

**Comment by kschrab** [comment]
> @schwepmo Libsumo currently does not work on Windows with SUMO 1.16.0. libsumojni.dll cannot be loaded :( I raised an issue here: [eclipse/sumo#12605](https://github.com/eclipse/sumo/issues/12605)
> 
> However, I tried out Barnim with SUMO 1.14.0 and 1.15.0 using the 1.16.0 jarfile and it succeeded, therefore I added 1.14.0 and 1.15.0 to the compatible versions in `LibSumoAmbassador.VALID_LIBSUMO_VERSIONS`

@schwepmo I've found a solution that runs with SUMO 1.16.0 but I would prefer to put this into a new PR. So from my side this PR could be merged now if you don't mind.

**Pull request merged**